### PR TITLE
[Fix #3019] Add Style/EmptyCaseCondition cop

### LINKED ISSUE #3019: Feature Request: Style/EmptyCaseCondition
Just ran into an unexpected usage of case statements in ruby, 

While this works, it's not the most grok-able thing:

```ruby
def assert_correct_copy(new_version, old_version)
  case
  when new_version.nil?
    # Weird
    raise "This should never happen -- please pass in non-nil"
  when new_version.to_json == old_version.to_json
    # Do nothing.
    log.info("Doing nothing, new_version == old_version")
  else
    actual_implementation(new_version, old_version)
  end
end
```

In reality, this picks the first truthy thing.

It'd be cleaner to rewrite as an if-statement:
```ruby
def assert_correct_copy(new_version, old_version)
  if new_version.nil?
    # Weird
    raise "This should never happen -- please pass in non-nil"
  elsif new_version.to_json == old_version.to_json
    # Do nothing.
    log.info("Doing nothing, new_version == old_version")
  else
    actual_implementation(new_version, old_version)
  end
end
```

Realistically, I'd love a style cop that checks for usages of `case` that switch on no variable.

That means:
```ruby
case foo # <- ok
when bar
# ...
```
```ruby
case # <- not ok
when self.can_do_action?
# ...
```

### PR DESCRIPTION
**Pull request by owst**
This adds a cop to catch case statements with an empty condition:
```
case
when 1 == 2:
  :foo
when 1 == 3:
  :bar
else
  :baz
end
```
and auto-corrects it to
```
if 1 == 2:
  :foo
elsif 1 == 3:
  :bar
else
  :baz
end
```

The subtle part of auto-correction was correctly handling:
```
case
when 1, 2, 3
  :foo
end
```
Which cannot just be re-written to:
```
if 1, 2, 3
  :foo
end
```
I chose instead to rewrite to:
```
if [1, 2, 3].any?
  :foo
end
```
Please let me know what you think of that choice, and the rest of the PR!

Before submitting a PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Used the same coding conventions as the rest of the project.
* [x] Feature branch is up-to-date with `master` (if not - rebase it)
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../CONTRIBUTING.md#changelog-entry-format).
* [x] All tests are passing.
* [x] The new code doesn't generate RuboCop offenses.
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.

[1]: http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html

### CODE REVIEW

File: lib/rubocop/cop/style/empty_case_condition.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
+# encoding: utf-8
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks for case statements with an empty condition.
+      class EmptyCaseCondition < Cop
+        MSG = 'Do not use empty case condition, instead use an if '\
```
Comment (bbatsov) [inline lib/rubocop/cop/style/empty_case_condition.rb]: Add backticks around `case` and `if`.

File: lib/rubocop/cop/style/empty_case_condition.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+              'statement.'.freeze
```
Comment (bbatsov) [inline lib/rubocop/cop/style/empty_case_condition.rb]: statement -> expression

File: lib/rubocop/cop/style/empty_case_condition.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
+# encoding: utf-8
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks for case statements with an empty condition.
```
Comment (bbatsov) [inline lib/rubocop/cop/style/empty_case_condition.rb]: Add some examples in the class's doc (see other cops).

File: lib/rubocop/cop/style/empty_case_condition.rb [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+          condition_node = case_node.children.first
+
+          add_offense(case_node, :keyword, MSG) if condition_node.nil?
+        end
+
+        private
+
+        def autocorrect(case_node)
+          lambda do |corrector|
+            when_nodes = case_node.children[1..-2]
```
Comment (bbatsov) [inline lib/rubocop/cop/style/empty_case_condition.rb]: I prefer using splatting to resorting to `children`.

**Comment by bbatsov** [comment]
This has to be rebased. Apart from my small remarks, the PR looks good.

**Comment by owst** [comment]
Thanks for the comments @bbatsov - all done!

File: lib/rubocop/cop/style/empty_case_condition.rb (hunk 3 above)
Comment (owst) [inline lib/rubocop/cop/style/empty_case_condition.rb]: Done

**Pull request merged**